library/ssl_tls.c

```diff
@@ -8041,6 +8073,11 @@ void mbedtls_ssl_free( mbedtls_ssl_context *ssl )
     mbedtls_free( ssl->cli_id );
 #endif
 
+#if defined(MBEDTLS_ZERO_RTT) && defined(MBEDTLS_SSL_SRV_C)
+    if( ssl->early_data_server_buf != NULL )
+        mbedtls_free( ssl->early_data_server_buf );
```

Comment:
We have to reliably zeroize this buffer to avoid a data leak. Use `mbedtls_platform_zeroize()` prior to the call to `free()`.